src/grants/GrantFund.sol

```
@@ -112,26 +112,33 @@ contract GrantFund is IGrantFund, ExtraordinaryFunding, StandardFunding {
         QuadraticVoter storage voter = quadraticVoters[currentDistribution.id][msg.sender];
         uint256 screeningStageEndBlock = _getScreeningStageEndBlock(currentDistribution);
 
-        // this is the first time a voter has attempted to vote this period
-        if (voter.votingPower == 0) {
-            voter.votingPower          = Maths.wpow(_getVotesSinceSnapshot(msg.sender, screeningStageEndBlock - VOTING_POWER_SNAPSHOT_DELAY, screeningStageEndBlock), 2);
-            voter.remainingVotingPower = voter.votingPower;
-        }
+        // check that the funding stage is active
+        if (block.number > screeningStageEndBlock && block.number <= currentDistribution.endBlock) {
```

do we want to inform user that something wrong here like
`if (block.number <= screeningStageEndBlock || block.number > currentDistribution.endBlock) revert FundingStageInactive()`
or just silently do nothing if funding stage inactive?

---

Yeah that might make sense - i'd been avoiding that to minimize the difference between the voteMulti and `_castVote` functions. Can't really do that revert in `_castVote` as it's needed to override the various voting methods in `Governor` and is the means by which the different voting mechanisms are selected

---

I am in favour of adding revert here as in the `_castVote` the voting would be done, either in screening or in funding. Here if we do not revert the transaction will pass but will not vote anywhere, revert would be a better option. 
I think adding revert will minimize the difference between voteMulti and _castVote more when compared with not adding a revert because then in both the cases if a transaction is successful that would mean voting was done.

---

The problem is `_castVote` needs to support every single voting mechanism. It isn't possible to revert in one of the cases as it could be meant to be a different stage or mechanism. We would have to do a wholesale rewrite to add it to `_castVote` which i'm not sure is possible while still overriding Governor. We could add it to the `multi` methods but they would then be inconsistent with singular votes done through `_castVote`